components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/internal/ElasticFunctionsServiceComponent.java

```diff
@@ -0,0 +1,163 @@
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+/**
+ * OSGi declarative services component which handle ELK related conditional auth functions.
+ */
+
+@Component(
+        name = "identity.conditional.auth.functions.elk.component",
+        immediate = true
+)
+public class ElasticFunctionsServiceComponent {
+
+    public static final String FUNC_CALL_ELASTIC = "callElastic";
+    public static final String SECURITY_TRUSTSTORE_LOCATION = "Security.TrustStore.Location";
+    public static final String SECURITY_TRUSTSTORE_TYPE = "Security.TrustStore.Type";
+    public static final String SECURITY_TRUSTSTORE_PASSWORD = "Security.TrustStore.Password";
+
+    private static final Log LOG = LogFactory.getLog(ElasticFunctionsServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+        try {
+            JsFunctionRegistry jsFunctionRegistry = ElasticFunctionsServiceHolder.getInstance().getJsFunctionRegistry();
+
+            CallElasticFunction callElastic = new CallElasticFunctionImpl();
+            jsFunctionRegistry.register(JsFunctionRegistry.Subsystem.SEQUENCE_HANDLER, FUNC_CALL_ELASTIC, callElastic);
+
+            BundleContext bundleContext = context.getBundleContext();
+            ElasticAnalyticsEngineConfigImpl analyticsFunctionConfig = new ElasticAnalyticsEngineConfigImpl();
+            bundleContext.registerService(IdentityConnectorConfig.class.getName(), analyticsFunctionConfig, null);
+
+            ServerConfigurationService config = ElasticFunctionsServiceHolder.getInstance()
+                    .getServerConfigurationService();
+
+            String filePath = config.getFirstProperty(SECURITY_TRUSTSTORE_LOCATION);
+            String keyStoreType = config.getFirstProperty(SECURITY_TRUSTSTORE_TYPE);
+            String password = config.getFirstProperty(SECURITY_TRUSTSTORE_PASSWORD);
+            try (InputStream keyStoreStream = Files.newInputStream(Paths.get(filePath))) {
+                KeyStore keyStore = KeyStore.getInstance(keyStoreType); // or "PKCS12"
+                keyStore.load(keyStoreStream, password.toCharArray());
+                ElasticFunctionsServiceHolder.getInstance().setTrustStore(keyStore);
+            } catch (IOException | CertificateException | KeyStoreException | NoSuchAlgorithmException e) {
+                LOG.error("Error while loading truststore.", e);
+            }
+        } catch (Throwable e) {
+            LOG.error("Error while activating AnalyticsFunctionsServiceComponent");
```

Comment:
Shall we log the thrown error here?

Reply:
Fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/111